src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CrlCache.cs

```diff
@@ -0,0 +1,253 @@
+                    // at least it can fail without using the network.
+                    //
+                    // If crl.NextUpdate is in the past, try downloading a newer version.
+                    DateTime nextUpdate = OpenSslX509CertificateReader.ExtractValidityDateTime(
+                        Interop.Crypto.GetX509CrlNextUpdate(crl));
+
+                    if (nextUpdate < verificationTime)
+                    {
+                        return false;
+                    }
+
+                    if (!Interop.libcrypto.X509_STORE_add_crl(store, crl))
+                    {
+                        //error:0B07D065:x509 certificate routines:X509_STORE_add_crl:cert already in hash table
+                        //throw Interop.libcrypto.CreateOpenSslCryptographicException();
+                    }
+
+                    return true;
+                }
+            }
+        }
+
+        private static void DownloadAndAddCrl(
+            X509Certificate2 cert,
+            SafeX509StoreHandle store,
+            ref TimeSpan remainingDownloadTime)
+        {
+            string url = GetCdpUrl(cert);
+
+            if (url == null)
+            {
+                return;
+            }
+
+            // X509_STORE_add_crl will increase the refcount on the CRL object, so we should still
+            // dispose our copy.
+            using (SafeX509CrlHandle crl = CertificateAssetDownloader.DownloadCrl(url, ref remainingDownloadTime))
+            {
+                // null is a valid return (e.g. no remainingDownloadTime)
+                if (crl != null && !crl.IsInvalid)
+                {
+                    if (!Interop.libcrypto.X509_STORE_add_crl(store, crl))
+                    {
+                        //error:0B07D065:x509 certificate routines:X509_STORE_add_crl:cert already in hash table
+                        //throw Interop.libcrypto.CreateOpenSslCryptographicException();
+                    }
+
+                    // It doesn't matter if saving it fails
+                    try
+                    {
+                        string crlFile = GetCachedCrlPath(cert, mkDir: true);
+
+                        using (SafeBioHandle bio = Interop.libcrypto.BIO_new_file(crlFile, "wb"))
+                        {
+                            if (!bio.IsInvalid)
+                            {
+                                Interop.libcrypto.PEM_write_bio_X509_CRL(bio, crl);
+                            }
+                        }
+                    }
+                    catch (IOException)
+                    {
+                    }
+                }
+            }
+        }
+        
+        private static string GetCachedCrlPath(X509Certificate2 cert, bool mkDir=false)
+        {
+            OpenSslX509CertificateReader pal = (OpenSslX509CertificateReader)cert.Pal;
+
+            string crlDir = PersistedFiles.GetUserFeatureDirectory("cryptography", "crls");
```

Comment:
Is there somewhere we could centralize the "cryptography" constant so that we don't accidentally introduce spelling mistakes in the various call sites?